cli/command/image/push.go

```diff
@@ -48,12 +52,30 @@ func NewPushCommand(dockerCli command.Cli) *cobra.Command {
 	flags.BoolVarP(&opts.all, "all-tags", "a", false, "Push all tags of an image to the repository")
 	flags.BoolVarP(&opts.quiet, "quiet", "q", false, "Suppress verbose output")
 	command.AddTrustSigningFlags(flags, &opts.untrusted, dockerCli.ContentTrustEnabled())
+	flags.StringVar(&opts.platform, "platform", os.Getenv("DOCKER_DEFAULT_PLATFORM"),
+		`Push a platform-specific manifest as a single-platform image to the registry.
+'local': Platform the cli is running on
+'remote': Platform the daemon is running on
+'os[/arch[/variant]]': Explicit platform (eg. linux/amd64)`)
+	flags.SetAnnotation("platform", "version", []string{"1.45"})
 
 	return cmd
 }
 
 // RunPush performs a push against the engine based on the specified options
 func RunPush(ctx context.Context, dockerCli command.Cli, opts pushOptions) error {
+	if opts.platform != "" {
+		if _, isTty := term.GetFdInfo(dockerCli.Err()); isTty {
+			_, _ = fmt.Fprint(dockerCli.Err(), "\x1b[1;37m\x1b[1;46m[ NOTE ]\x1b[0m\x1b[0m ")
+		} else {
+			_, _ = fmt.Fprint(dockerCli.Err(), "[ NOTE ] ")
+		}
+		_, _ = fmt.Fprintln(dockerCli.Err(), `Selecting a single platform for the push operation will push the image manifest for that platform only.
+This won't push the image index/manifest list which means that other components like Buildkit attestations won't be pushed.
+If you want to only push a single platform while preserving the attestations, please build an image with only that platform and push it instead.`)
```

Comment:
Related to the above, after talking about manifests manifest lists and indexes we say "single platform"

Reply:
The problem is that "single platform" isn't always a 1:1 mapping between an index and a single manifest.

Is the index below a single platform image or a multi-platform image?

```json
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.oci.image.index.v1+json",
  "manifests": [
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "digest": "sha256:34b7d4a2f050f8a9077fd435b3b1778e091af743f0f4c8c47d109cfda47b0c48",
      "size": 480,
      "platform": {
        "architecture": "arm64",
        "os": "linux"
      }
    },
    {
      "mediaType": "application/vnd.oci.image.manifest.v1+json",
      "digest": "sha256:a7931924962b793438f641b320d496ebca34968e850f7b8d7a5ea59dc88283cc",
      "size": 565,
      "annotations": {
        "vnd.docker.reference.digest": "sha256:34b7d4a2f050f8a9077fd435b3b1778e091af743f0f4c8c47d109cfda47b0c48",
        "vnd.docker.reference.type": "attestation-manifest"
      },
      "platform": {
        "architecture": "unknown",
        "os": "unknown"
      }
    }
  ]
}
```